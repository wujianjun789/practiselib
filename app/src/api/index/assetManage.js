/**
 * Created by a on 2017/7/11.
 */
import {getHttpHeader, httpRequest} from '../../util/network'

export function getAssetManageModule(cb) {
    let headers = getHttpHeader();
    httpRequest('/', {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}

export function getDeviceProperty(cb) {
    let headers = getHttpHeader();
    httpRequest('/', {
        headers: headers,
        method: 'GET'
    }, response=>{
        cb && cb(response);
    })
}