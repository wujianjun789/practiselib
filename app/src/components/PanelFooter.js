import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
/**
 * Panel footer with button
 * @prop funcNames {array}  btn's function
 * @prop btnTitles {array}  btn's title
 */
export default class PanelFooter extends PureComponent {
    render() {
        const {funcNames, btnTitles, btnDisabled=[false, false], btnClassName=['btn-default','btn-primary']} = this.props;
        return (
            <div className="row">
                <div className="btn-toolbar pull-right padding-right" role="toolbar" aria-label="toolbar">
                    <div className="btn-group" role='group' aria-label="group">
                        <button type="button" className={`btn ${btnClassName[0]}`} disabled={btnDisabled[0]} onClick={this.props[funcNames[0]]}>{btnTitles[0]}</button>
                    </div>
                    <div className="btn-group" role="group" aria-label="group">
                        <button type="button" className={`btn ${btnClassName[1]}`} disabled={btnDisabled[1]} onClick={this.props[funcNames[1]]}>{btnTitles[1]}</button>
                    </div>
                </div>
            </div>
        )
    }
}

PanelFooter.PropTypes = {
    funcNames: PropTypes.arrayOf(PropTypes.string).isRequired,
    btnTitles: PropTypes.arrayOf(PropTypes.string).isRequired,
    btnDisabled: PropTypes.arrayOf(PropTypes.bool)
}