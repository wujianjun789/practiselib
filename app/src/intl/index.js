import enLocaleData from 'react-intl/locale-data/en';
import zhLocaleData from 'react-intl/locale-data/zh';
import { addLocaleData } from 'react-intl';
import { getLocalStorage, setLocalStorage} from '../util/cache'
addLocaleData(enLocaleData);
addLocaleData(zhLocaleData);
const intl = {
    defaultLocale: 'zh',
}
function getIntl(lang) {
    intl.locale = lang.locale;
    intl.messages = lang.messages;
    return intl;
}
export function getDefaultIntl(cb) {
    let appLanguage = getLocalStorage("appLanguage");
    let newLanguage = appLanguage || 'zh';
    setLocalStorage("appLanguage", newLanguage);
    getTargetIntl(newLanguage, cb);
}

export function getIntlEn(cb) {
    if(typeof require.ensure === "function"){
        require.ensure(['./en'], () => {
            const en = require('./en').default;
            cb(getIntl(en));
        }, 'intl.en')
    }
}
export function getIntlZh(cb) {
    if(typeof require.ensure === "function"){
        require.ensure(['./zh'], () => {
            const zh = require('./zh').default;
            cb(getIntl(zh));
        }, 'intl.zh')
    }
}

export function getTargetIntl(locale, cb) {
    switch (locale) {
        case 'zh':
            return getIntlZh(cb);
        default:
            return getIntlEn(cb);
    }
}

