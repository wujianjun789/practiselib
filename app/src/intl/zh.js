export default {
    locale: 'zh',
    messages: {
        //公共
        'name':'名称',
        'id':'编号',

        //按钮
        ' ':' ',
        'button.add': '添加',
        'button.edit': '编辑',
        'button.delete': '删除',
        'button.confirm': '确认',
        'button.import': '导入',        
        'button.cancel': '取消',
        'button.set': '设定',
        'button.finish': '完成',

        //地图
        'map.location':'地图位置',

        //提示
        'app.not.found':'没有找到结果',

        //删除弹框
        'delete.user':'是否删除选中用户？',
        'delete.device':'是否删除选中设备？',        

        //首页
        'app.title':'智慧路灯管理系统',
        'app.asset.manage':'资产管理',
        'app.permission.manage':'权限管理',
        'app.system.operation':'系统运维',
        'app.smart.light':'智慧路灯',
        'app.light':'智能照明',
        'app.report.manage':'报表管理',
        'app.mediaPublish':'媒体发布',
        'app.visualization':'可视化',
        'app.domain.manage':'域管理',

        //资产管理
        'asset.model': '资产管理',
        'asset.statistics': '资产统计',

        //权限管理
        'permission.username':'用户名称',
        'permission.lastLoginDate':'最后登录时间',
        'permission.admin':'系统管理员',
        'permission.deviceAdmin':'设备管理员',
        'permission.deviceOperator':'设备操作员',
        'permission.guest':'访客',
        'permission.addUser':'添加用户',
        'permission.userData':'用户资料',
        'permission.domain':'域权限',
        'permission.module':'模块权限',
        'permission.input.username':'请输入用户名称',
        'permission.input.domain':'请输入域名称',        
        'permission.added':'已添加',
        'permission.lastname':'姓氏',
        'permission.firstname':'名字',
        'permission.password':'密码',
        'permission.repassword':'重复密码',
        'permission.grade':'用户等级',

        //系统运维
        'sysOperation.device.config':'设备配置',
        'sysOperation.selected.device':'选中设备',
        'sysOperation.whiteList':'白名单',
        'sysOperation.include':'包含',
        'sysOperation.devices':'个设备',
        'sysOperation.input.device':'输入设备名称',
        'sysOperation.domain':'域',
        'sysOperation.device':'设备名称',
        'sysOperation.type':'型号',
        'sysOperation.id':'设备编号',
        'sysOperation.lng':'经度',
        'sysOperation.lat':'纬度',
        'sysOperation.dataOrigin':'数据源',
        'sysOperation.dataOrigins':'个数据源',
        'sysOperation.gateway':'网关',
        'sysOperation.lamp.control':'单灯控制器',
        'sysOperation.pole':'灯杆',
        'sysOperation.screen':'显示屏',
        'sysOperation.sensor':'传感器',
        'sysOperation.xes':'数据采集仪',
        'sysOperation.charge':'充电桩',
        'sysOperation.addDevice':'添加设备',
        'sysOperation.importList':'导入列表',
        'sysOperation.select':'选择列表文件路径',
        'sysOperation.cover':'覆盖已有设备',
        'sysOperation.input.asset':'请输入素材名称',
        'sysOperation.select.sensor':'选择传感器：',
        'sysOperation.label':'标签：',
        'sysOperation.input.id':'输入编号添加',

        //智慧路灯
        'app.map': '地图',
        'app.list': '列表',
        'app.control': '控制',
        'app.scene': "场景",
        'app.strategy': "策略",

        'app.search.placeholder.name': "搜索名称或域",
        'app.search.list': '返回',
        'app.device': '设备',
        'app.domain': '域',
        'app.open': '开',
        'app.close': '关',
        'app.device.control': '设备控制',
        'app.apply': '应用',
        'app.environment-brightness-control': '环境亮度控制',
        'app.manual.dimming': '手动调光',
        'app.strategy.dimming': '策略调光',
        'app.group.dimming': '组调光',
        'app.switch': '切换',
        'app.preset': '预置',
        'app.zoom': '变焦',

        'app.schedule':'时间表',
        'app.led.control': '显示屏控制',
        'app.online': '在线',
        'app.offline': '离线',
        'app.charging': '充电中。。。',
        'app.charging.fault': '充电故障',

        'app.manual': '手动',

        'app.scene.order': '场景序号',
        'app.device.number': '设备数量',
        'app.input.scene.name': '输入场景名称',

        'app.time.strategy':'时间策略',
        'app.sensor.strategy':'传感器策略',
        'app.latlng.strategy':'经纬度策略',
        'app.input.strategy.name':'输入策略名称',
        'app.strategy.name':'策略名称',
        'app.strategy.type':'策略类型',
        'app.device.type':'设备种类',
        'app.set.state':'设定状态',
        'app.strategy.param':'策略参数',
        'app.include.target.devices':'包含目标设备',

        'app.input.device.name':'请输入设备名称',
        'app.device.name': '设备名称',
        'app.online.state': '在线状态',
        'app.fault.state': '故障状态',
        'app.current.brightness':'当前亮度',
        'app.brightness.mode':'亮度模式',
        'app.update.time':'更新时间',
        'app.light.state':'灯状态',
        'app.switch.state':'开关状态',
        'app.brightness':'亮度',
        'app.voltage':'电压',
        'app.electric.current':'电流',
        'app.power':'功率',

        'app.device.operation':'设备操作',
        'app.device.switch':'设备开关',
        'app.dimming': '调光',
        
        //以前的
        'app.link.index': '首页',
        'app.link.config': '配置',
        'app.search': '搜索',
        'app.search.smartlight': '智慧路灯',
        'app.search.camera': '摄像头',
        'app.search.led': '屏',
        'app.screen.state': '屏状态',
        'app.environment.state': '环境状态',
        'app.camera.control': '摄像头控制',

        'app.led.status': '显示屏离线',
        'app.refresh': '刷新',

        'app.publish': '发布',
        'app.force': '焦距',

        'app.work': '激活',
        'app.call': '调用',
        'app.plus': '加',
        'app.minus': '减',
        'app.left-top': '左上',
        'app.top': '上',
        'app.right-top': '右上',
        'app.left': '左',
        'app.center': '中',
        'app.right': '右',
        'app.left-bottom': '左下',
        'app.bottom': '下',
        'app.right-bottom': '右下',
        'app.screen.refresh': '单击刷新',

        'app.width': '宽度',
        'app.height': '高度',
        'app.version': '版本号',
        'app.brightness_mode': '亮度模式',
        'app.brightness': '当前亮度',
        'app.sys_fault': '系统故障',
        'app.vram_fault': 'VRAM故障',
        'app.disp_module_fault': '显示模组故障',
        'app.disp_module_power_fault': '显示模组电源故障',
        'app.single_pixel_tube_fault': '单像素管故障',
        'app.detection_sys_fault': '检测系统故障',
        'app.ac_fault': '交流电故障',
        'app.lightning_arrester_fault': '防雷器故障',
        'app.photosensor_fault': '光敏部件故障',
        'app.abnormal_temperature_fault': '温度异常故障',
        'app.door_switch_fault': '门开关故障',
        'app.fault': '故障',
        'app.fault_info': '屏故障信息',
        
        'app.normal': '正常',
        'app.abnormal': '异常',

        'app.temperature': '环境温度',
        'app.humidity': '环境湿度',
        'app.air-pressure': '气压',
        'app.noise': '噪音',
        'app.wind-speed': '风速',
        'app.wind-direction': '风向',
        'app.pm25': 'PM2.5',
        'app.o2': 'O2',
        'app.co': 'CO',

        'app.Static': '无风',
        'app.N': '北风',
        'app.NE': '东北风',
        'app.E': '东风',
        'app.SE': '东南风',
        'app.S': '南风',
        'app.SW': '西南风',
        'app.W': '西风',
        'app.NW': '西北风',

        'app.screen.content':'上传播放列表需要一段时间，',
        'app.screen.content2':'请您耐心等待，请稍后查看。',
        'app.confirm':'确定',

        'app.alarm':'警报',
        'app.name':'名称',
        'app.type':'类型',
        'app.date':'时间',

        'manage.title':'StarRiver 管理',

        'manage.link.gateway': '网关',

        'manage.link.whiteList': '白名单',
        'manage.link.camera': '摄像头',
        'manage.link.smartLight': '智慧路灯',
        'manage.link.screen': '屏',
        'manage.link.playList':'播放列表',
        'manage.link.rfid':'RFID',

        'panel.info': '无相关数据！',
        'manage.name':'名称',

        'manage.screen.table.esn': 'ESN',
        'manage.screen.table.name': '名称',
        'manage.screen.table.domain': '域',
        'manage.smartlight.table.name': '名称',
        'manage.smartlight.table.screen': '显示屏',
        'manage.smartlight.table.camera': '摄像头',
        'manage.smartlight.table.rfid': 'RFID',
        'manage.smartlight.table.longitude': '经度',
        'manage.smartlight.table.langitude': '纬度',

        'manage.popup.pleaseSelect': '请选择',
        'manage.popup.confirmDelete': '确认删除',
        'manage.popup.tips': '提示',
        'manage.popup.smartlight.addSmartlight': '新增智慧路灯',
        'manage.popup.smartlight.editSmartlight': '编辑智慧路灯',
        'manage.popup.smartlight.pole': '灯杆',
        'manage.popup.smartlight.screen': '显示屏',
        'manage.popup.smartlight.camera': '摄像头',
        'manage.popup.smartlight.rfid': 'RFID',
        'manage.popup.screen.addScreen': '新增显示屏',
        'manage.popup.screen.editScreen': '编辑显示屏',
        'manage.popup.screen.esn': 'ESN',
        'manage.popup.screen.name': '名称',
        'manage.popup.screen.domain': '域',

        'manage.playlist.time':'修改时间',
        'manage.playlist.add':'新增播放列表',
        'manage.playlist.browse':'选择文件',

        'manage.camera.ip':'IP',
        'manage.camera.rtsp':'RTSP端口',
        'manage.camera.http':'HTTP端口',
        'manage.camera.username':'用户名',
        'manage.camera.password':'密码',
        'manage.camera.channel':'信道',
        'manage.camera.add':'新增摄像头',
        'manage.camera.edit':'编辑摄像头',

        'manage.rfid.esn': 'ESN',
        'manage.rfid.name': '名称',
        'manage.rfid.domain': '域',
        'manage.rfid.add':'新增RFID',
        'manage.rfid.edit':'编辑RFID',
    }
}