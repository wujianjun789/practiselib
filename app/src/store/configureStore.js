import { createStore, applyMiddleware, compose } from 'redux';
import rootReducer from '../common/reducer';
import thunk from 'redux-thunk';


export default function configureStore(initialState) {

    const composeEnhancers =
        process.env.NODE_ENV !== 'production' &&
            typeof window === 'object' &&
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ ?
            window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
                // Specify here name, actionsBlacklist, actionsCreators and other options
            }) : compose;

    const enhancer = composeEnhancers(
        applyMiddleware(thunk),
        // other store enhancers if any
    );
    const store = createStore(rootReducer, initialState, enhancer);
    return store;

}
