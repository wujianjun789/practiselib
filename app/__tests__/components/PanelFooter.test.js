import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import {Provider} from 'react-redux';
import {IntlProvider} from 'react-intl-redux';
import configureStore from '../../src/store/configureStore';

import PanelFooter from '../../src/components/PanelFooter';
import {getDefaultIntl} from '../../src/intl/index'

getDefaultIntl();
const store = configureStore();
describe('<PanelFooter />', () => {
    it('render with funcNames=["onCancel","onConfirm"], btnTitles=["Cancel","Confirm"]', () => {
        const root = mount(<Provider store={store}>
                <IntlProvider>
                    <PanelFooter funcNames={["onCancel","onConfirm"]} btnTitles={["Cancel","Confirm"]} />
                </IntlProvider>
            </Provider>);
        const cmp = root.find("PanelFooter");

        const btnCancel = cmp.find('.btn.btn-default');
        // expect(btnCancel.length).toBe(1);
        // expect(btnCancel.prop('disabled')).not.toBeTruthy();
        // expect(btnCancel.text()).toBe('Cancel');

        const btnConfirm = cmp.find('.btn.btn-primary');
        // expect(btnConfirm.length).toBe(1);
        // expect(btnConfirm.prop('disabled')).not.toBeTruthy();
        // expect(btnConfirm.text()).toBe('Confirm');
    });

    it('render with funcNames=["onCancel","onConfirm"], btnTitles=["Cancel","Confirm"],btnDisabled={[true, true]},btnClassName={["btn-custom1","btn-custom2"]}' , () => {
        const root = shallow(<Provider store={store}>
            <IntlProvider>
                <PanelFooter funcNames={["onCancel","onConfirm"]} btnTitles={["Cancel","Confirm"]} />
            </IntlProvider>
        </Provider>);
        const cmp = root.find('PanelFooter');
        const btnCancel = cmp.find('.btn.btn-custom1');
        // expect(btnCancel.length).toBe(1);
        // expect(btnCancel.prop('disabled')).toBeTruthy();
        // expect(btnCancel.text()).toBe('Cancel');

        const btnConfirm = cmp.find('.btn.btn-custom2');
        // expect(btnConfirm.length).toBe(1);
        // expect(btnConfirm.prop('disabled')).toBeTruthy();
        // expect(btnConfirm.text()).toBe('Confirm');
    });

    it('btn click', () => {
        const click = jest.fn();
        const root = shallow(<Provider store={store}>
            <IntlProvider>
                <PanelFooter funcNames={["onCancel","onConfirm"]} btnTitles={["Cancel","Confirm"]} />
            </IntlProvider>
        </Provider>);
        const cmp = root.find("PanelFooter");
        const btnCancel = cmp.find('.btn.btn-default');
        // btnCancel.simulate('click');
        // expect(click).toHaveBeenCalledTimes(1);

        const btnConfirm = cmp.find('.btn.btn-primary');
        // btnConfirm.simulate('click');
        // expect(click).toHaveBeenCalledTimes(2);
    });

    it('snapshot with funcNames=["onCancel","onConfirm"], btnTitles=["Cancel","Confirm"]', () => {
        const cmp = renderer.create(<Provider store={store}>
            <IntlProvider>
                <PanelFooter funcNames={["onCancel","onConfirm"]} btnTitles={["Cancel","Confirm"]} />
            </IntlProvider>
        </Provider>);
        let tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('snapshot with funcNames=["onCancel","onConfirm"], btnTitles=["Cancel","Confirm"],btnDisabled={[true, true]},btnClassName={["btn-custom1","btn-custom2"]}', () => {
        const cmp = renderer.create(<Provider store={store}>
            <IntlProvider>
                <PanelFooter funcNames={["onCancel","onConfirm"]} btnTitles={["Cancel","Confirm"]} />
            </IntlProvider>
        </Provider>);
        let tree = cmp.toJSON();
        expect(tree).toMatchSnapshot();
    })
})