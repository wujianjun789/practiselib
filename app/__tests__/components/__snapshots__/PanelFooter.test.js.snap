exports[`<PanelFooter /> snapshot with funcNames=["onCancel","onConfirm"], btnTitles=["Cancel","Confirm"] 1`] = `
<div
  className="modal-footer">
  <button
    className="btn btn-default"
    disabled={false}
    onClick={undefined}
    type="button">
    Cancel
  </button>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={undefined}
    type="button">
    Confirm
  </button>
</div>
`;

exports[`<PanelFooter /> snapshot with funcNames=["onCancel","onConfirm"], btnTitles=["Cancel","Confirm"],btnDisabled={[true, true]},btnClassName={["btn-custom1","btn-custom2"]} 1`] = `
<div
  className="modal-footer">
  <button
    className="btn btn-custom1"
    disabled={true}
    onClick={undefined}
    type="button">
    Cancel
  </button>
  <button
    className="btn btn-custom2"
    disabled={true}
    onClick={undefined}
    type="button">
    Confirm
  </button>
</div>
`;
